From 6b82a74ffcabfd8a6413ceee794fc9bc367fae84 Mon Sep 17 00:00:00 2001
From: nikesh <Nikesh.Oswal@wolfsonmicro.com>
Date: Wed, 8 Jan 2014 11:37:39 +0000
Subject: [PATCH 12/14] ASOC: RPi: Set WM8804 MCLK_DIV=0 (256fs)

Avoids any issues with PLL setup whilst switching between
RX and TX modes

Signed-off-by: nikesh <Nikesh.Oswal@wolfsonmicro.com>
---
 sound/soc/bcm/rpi-wolfson-sound-pi.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sound/soc/bcm/rpi-wolfson-sound-pi.c b/sound/soc/bcm/rpi-wolfson-sound-pi.c
index 7976d98..6c4dd79 100644
--- a/sound/soc/bcm/rpi-wolfson-sound-pi.c
+++ b/sound/soc/bcm/rpi-wolfson-sound-pi.c
@@ -497,6 +497,13 @@ static int snd_rpi_wsp_config_8804_clks(struct snd_soc_codec *wm8804_codec,
 		return ret;
 	}
 
+	/*Fix MCLKDIV=0 for 256fs to avoid any issues switching between TX and RX. RX always expects 256fs*/
+	ret = snd_soc_dai_set_clkdiv(wm8804_dai, WM8804_MCLK_DIV, 0 );
+	if (ret != 0) {
+		dev_err(wm8804_codec->dev, "Failed to set MCLK_DIV to 256fs: %d\n", ret);
+		return ret;
+	}
+
 	/*Set CLKOUT as OSC Frequency*/
 	ret = snd_soc_dai_set_sysclk(wm8804_dai, WM8804_CLKOUT_SRC_OSCCLK, WM8804_CLKOUT_HZ, 0);
 	if (ret != 0) {
-- 
1.7.9.5

